CMakeLists.txt

```diff
@@ -190,32 +189,32 @@ if(COSE_C_USE_MBEDTLS)
   if(COSE_C_USE_FIND_PACKAGE)
     find_package(MbedTLS REQUIRED)
   else()
-    externalproject_add(
+    include(FetchContent)
+    fetchcontent_declare(
       project_mbedtls
       GIT_REPOSITORY https://github.com/ARMmbed/mbedtls
-      CMAKE_ARGS -DENABLED_PROGRAMS=OFF
-                 -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
-                 -DUSE_SHARED_MBEDTLS_LIBRARY=${BUILD_SHARED_LIBS}
-                 -DENABLE_TESTING=OFF
-                 -DLIB_INSTALL_DIR=${install_dir}/lib
-                 INTALL_DIR
-                 "${dist_dir}"
-      UPDATE_DISCONNECTED 1)
-
-    externalproject_get_property(project_mbedtls install_dir)
-    include_directories("${install_dir}/include")
-    add_library(mbedtls STATIC IMPORTED)
-    if(MSVC)
-      set_property(TARGET mbedtls PROPERTY IMPORTED_LOCATION
-                                           "${dist_dir}/lib/${CMAKE_SHARED_MODULE_PREFIX}mbedcrypto.lib")
-    else()
-      set_property(
-        TARGET mbedtls PROPERTY IMPORTED_LOCATION
-                                "${dist_dir}/lib/${CMAKE_SHARED_MODULE_PREFIX}mbedcrypto${CMAKE_SHARED_LIBRARY_SUFFIX}")
+      GIT_TAG master)
+
+    fetchcontent_getproperties(project_mbedtls)
+    if(NOT project_mbedtls_POPULATED)
+      fetchcontent_populate(project_mbedtls)
+      set(ENABLED_PROGRAMS
+          OFF
+          CACHE BOOL "" FORCE)
+      set(USE_STATIC_MBEDTLS_LIBRARY
+          ${BUILD_STATIC_LIBS}
+          CACHE BOOL "" FORCE)
+      set(USE_SHARED_MBEDTLS_LIBRARY
+          ${BUILD_SHARED_LIBS}
+          CACHE BOOL "" FORCE)
+      set(ENABLE_TESTING
+          OFF
+          CACHE BOOL "" FORCE)
+      add_subdirectory(${project_mbedtls_SOURCE_DIR} ${project_mbedtls_BINARY_DIR} EXCLUDE_FROM_ALL)
     endif()
-    add_dependencies(mbedtls project_mbedtls)
+    # TODO: remove next line after https://github.com/ARMmbed/mbedtls/pull/2090
+    include_directories("${project_mbedtls_BINARY_DIR}/include")
```

Comment:
we need to add the include directory manually as upstream is still not using modern cmake targets

ref: https://github.com/ARMmbed/mbedtls/pull/2090